### PR TITLE
8313552: Fix -Wconversion warnings in JFR code

### PR DESCRIPTION
**Pull request by mgronlun**
Greetings,

this change set removes -Wconversion warnings in the JFR code.

Testing: jdk_jfr

Thanks
Markus

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8313552](https://bugs.openjdk.org/browse/JDK-8313552): Fix -Wconversion warnings in JFR code (**Enhancement** - P4)

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [9fa7a46e](https://git.openjdk.org/jdk/pull/15152/files/9fa7a46efd94f020cfb78b4ca29ddeb4cf930a30)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15152/head:pull/15152` \
`$ git checkout pull/15152`

Update a local copy of the PR: \
`$ git checkout pull/15152` \
`$ git pull https://git.openjdk.org/jdk.git pull/15152/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15152`

View PR using the GUI difftool: \
`$ git pr show -t 15152`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15152.diff">https://git.openjdk.org/jdk/pull/15152.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15152#issuecomment-1665429047)

### CODE REVIEW

File: src/hotspot/share/jfr/instrumentation/jfrEventClassTransformer.cpp [hunk 1]
```diff
@@ -909,9 +913,9 @@ static void adjust_stack_map(JfrBigEndianWriter& writer,
     writer.write<u1>(stream.get_u1(THREAD));
   }
 
-  u4 stack_map_attrib_len = writer.current_offset() - stack_map_attrib_len_offset;
+  u4 stack_map_attrib_len = static_cast<u4>(writer.current_offset() - stack_map_attrib_len_offset);
   // the stack_map_table_attributes_length value is exclusive
-  stack_map_attrib_len -= sizeof(u4);
```
Comment (coleenp) [inline src/hotspot/share/jfr/instrumentation/jfrEventClassTransformer.cpp:914]: I'm thinking we should have globalDefinitions.hpp:

```
  const int u4_size = (int)sizeof(u4);
  const int u2_size = (int)sizeof(u2);
  const int u1_size = (int)sizeof(u1);
```

I don't think you have to static_cast<u4>(4) here.

File: src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdBits.hpp [hunk 2]
```diff
@@ -35,28 +34,28 @@ class JfrTraceIdBits : AllStatic {
...
-  static void clear(jbyte bits, const T* ptr);
+  static void clear(uint8_t bits, const T* ptr);
 
   template <typename T>
-  static void clear_cas(jbyte bits, const T* ptr);
+  static void clear_cas(uint8_t bits, const T* ptr);
 
   template <typename T>
-  static void meta_clear(jbyte bits, const T* ptr);
+  static void meta_clear(uint8_t bits, const T* ptr);
```
Comment (coleenp) [inline src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdBits.hpp:58]: jbyte -> uint8_t is definitely preferable.

**Review by coleenp — COMMENTED** [review]
Thank you so much for fixing these warnings. I had one or two suggestions for changes.

File: src/hotspot/share/jfr/instrumentation/jfrEventClassTransformer.cpp (hunk 1 above)
Comment (mgronlun) [reply src/hotspot/share/jfr/instrumentation/jfrEventClassTransformer.cpp:914]: Since 4 is an int by default, the cast is needed.

Comment (mgronlun) [reply src/hotspot/share/jfr/instrumentation/jfrEventClassTransformer.cpp:914]: A constant value is fine.

File: src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdBits.hpp (hunk 2 above)
Comment (mgronlun) [reply src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdBits.hpp:58]: Yes.

File: src/hotspot/share/jfr/periodic/jfrPeriodic.cpp [hunk 3]
```diff
@@ -433,8 +433,8 @@ TRACE_REQUEST_FUNC(GCTLABConfiguration) {
 TRACE_REQUEST_FUNC(GCSurvivorConfiguration) {
   GCSurvivorConfiguration conf;
   EventGCSurvivorConfiguration event;
-  event.set_maxTenuringThreshold(conf.max_tenuring_threshold());
-  event.set_initialTenuringThreshold(conf.initial_tenuring_threshold());
+  event.set_maxTenuringThreshold(static_cast<u1>(conf.max_tenuring_threshold()));
+  event.set_initialTenuringThreshold(static_cast<u1>(conf.initial_tenuring_threshold()));
```
[the remark this answers is not in the document]
Comment (mgronlun) [reply src/hotspot/share/jfr/periodic/jfrPeriodic.cpp]: Ok, I have updated to use checked_cast.

**Review by coleenp — APPROVED** [review]
Looks good.  Thank you for fixing all of these.

**Pull request closed without merging**